### PR TITLE
Npcsystem changes

### PR DESCRIPTION
**Pull request by brunominervino**

### CODE REVIEW

**Comment by WibbenZ** [comment]
Regarding the loops, #1469 
By Mark
```
Yes, using ipairs is slower, but not as slow as the page you're linking to suggests (it's 4 years old though, so maybe it was accurate that long ago). In my tests, it was 8% slower with LuaJIT, and 20% slower with Lua, but it's a matter of nanoseconds, so it doesn't matter, we go for readability in this case. It takes millions of iterations before it's a matter of milliseconds. Whatever you're going to do inside the loop is going to dominate the total time anyway. The last paragraph under "Other Considerations" on that page is on point.
```

**Comment by brunominervino** [comment]
@WibbenZ, the loops were reverted.
Thank you :+1:

File: data/actions/scripts/quests/annihilator.lua [hunk 1]
```diff
@@ -15,7 +15,7 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	if item.itemid == 1945 then
 		local players = {}
 		for _, position in ipairs(playerPosition) do
-			local topPlayer = Tile(position):getTopCreature()
+			local topPlayer = position:getTile():getTopCreature()
```
Comment (WibbenZ) [inline data/actions/scripts/quests/annihilator.lua]: Revert this, no need to go through another function when you can just use Tile(position).

File: data/movements/scripts/tiles.lua [hunk 2]
```diff
@@ -25,7 +25,7 @@ function onStepIn(creature, item, position, fromPosition)
 	if Tile(position):hasFlag(TILESTATE_PROTECTIONZONE) then
 		local lookPosition = player:getPosition()
 		lookPosition:getNextPosition(player:getDirection())
-		local depotItem = Tile(lookPosition):getItemByType(ITEM_TYPE_DEPOT)
+		local depotItem = lookPosition:getTile():getItemByType(ITEM_TYPE_DEPOT)
```
Comment (WibbenZ) [inline data/movements/scripts/tiles.lua]: Revert this, no need to go through another function when you can just use Tile(position).

**Comment by WibbenZ** [comment]
FYI you have to remove them anyways since Mark removed that function in 4da1360db922fd8d7adce06300c266741e4fadfc

**Comment by brunominervino** [comment]
@WibbenZ, done!
Thank you again :+1:

**Comment by WibbenZ** [comment]
Good, you can most likely squash your commits and rename the PR to a more accurate name.

**Comment by brunominervino** [comment]
Done :)

**Comment by luanlucian93** [comment]
change too getItemName(itemid) to ItemType(itemId):getName(), please

**Comment by brunominervino** [comment]
Thank you @luanlucian93

**Comment by luanlucian93** [comment]
change too isItemFluidContainer(itemid) to ItemType(itemid):isFluidContainer()

**Pull request closed without merging**